### PR TITLE
Feature: model mode switching & command update

### PR DESCRIPTION
**Pull request by krautech**
## Description

Adds mode to models as well as renames and remaps associates commands with fallbacks.

### What changed?
`[scanner]`
- renamed `calibration_method` to `mode` (calibration_method is still available)

`[scanner model NAME]`
- added `model_mode: scan/touch` (based on whether youre using touch or not)

### Command Changes

`PROBE_SWITCH` will now change what mode you're in, either scan or touch and will ask you to `SAVE_CONFIG` in order to update `mode` under [scanner]

**Previous:** `CARTOGRAPHER_TOUCH CALIBRATE=1` (still works)
**Now:** `CARTOGRAPHER_CALIBRATE`

---------------------------

**Previous:** `CARTOGRAPHER_TOUCH METHOD=manual` (still works)
**Now:** `CARTOGRAPHER_CALIBRATE METHOD=manual`

## Checklist

The following relevant macros have been tested:

- [x] `CARTOGRAPHER_CALIBRATE`
- [x] `PROBE`
- [x] `PROBE_ACCURACY`
- [x] `CARTOGRAPHER_THRESHOLD_SCAN`
- [x] `CARTOGRAPHER_TOUCH CALIBRATE=1`
- [x] `CARTOGRAPHER_TOUCH`
- [x] `CARTOGRAPHER_TOUCH METHOD=manual`
- [x] `BED_MESH_CALIBRATE`

### CODE REVIEW

File: scanner.py [hunk 1]
```diff
@@ -280,6 +280,7 @@ def __init__(self, config):
         for sensor in [self.sensor, self.sensor_alt]:
             if sensor:  # Ensure the sensor is not None
                 sensor_name = sensor.upper()
+                self.sensor_name = sensor_name
```
Comment (Jomik) [inline scanner.py:283]: Shouldn't we just do this outside the loop?
Doesn't really matter, but just cleaner I think 😄

**Review by Jomik — DISMISSED** [review]

File: scanner.py (hunk 1 above)
Comment (krautech) [reply scanner.py:283]: Got a proposal?

Comment (Jomik) [reply scanner.py:283]: I think it should just be above line 280.
`self.sensor_name = self.sensor.upper()` :)

**Review by Jomik — APPROVED** [review]

File: scanner.py (hunk 1 above)
Comment (krautech) [reply scanner.py:283]: Needs to make sure `sensor` exists first though?

**Pull request merged**